src/test/test-mountpoint-util.c

```diff
@@ -57,7 +57,14 @@ static void test_mnt_id(void) {
                 assert_se(r > 0);
 
                 assert_se(sscanf(line, "%i %*s %*s %*s %ms", &mnt_id, &path) == 2);
-
+#if HAS_FEATURE_MEMORY_SANITIZER
+                /* We don't know the length of the string, so we need to unpoison it one char at a time */
+                for (const char *c = path; ;c++) {
+                        msan_unpoison(c, 1);
+                        if (!*c)
+                                break;
+                }
```

Comment:
maybe turns this into a static inline function msan_unpoison_string() to be placed next to our msan_unpoison() macro?

Reply:
I'd prefer to keep it simple for now. If we find another spot where this should be used, it'll be simple to split out the function.